client/components/goals.jsx

```diff
@@ -0,0 +1,105 @@
+import React from 'react';
+import AppContext from '../lib/app-context';
+import GoalsForm from './goals-form';
+import Goal from './individual-goal';
+
+export default class Goals extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      goals: []
+    };
+    this.addGoal = this.addGoal.bind(this);
+    this.toggleCompleted = this.toggleCompleted.bind(this);
+  }
+
+  componentDidMount() {
+    fetch(`/api/goals/${this.context.user.userId}`)
+      .then(res => res.json())
+      .then(goals => this.setState({
+        goals
+      }))
+      .catch(err => console.error(err));
+  }
+
+  addGoal(newGoal) {
+    const req = {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(newGoal)
+    };
+    fetch(`/api/goals/${this.context.user.userId}`, req)
+      .then(res => res.json())
+      .then(goal => {
+        this.state.goals.push(goal);
+        this.setState({
+          goals: this.state.goals
+        });
+      })
+      .catch(err => console.error(err));
+  }
+
+  toggleCompleted(goalId) {
+    const { goals } = this.state;
+    const target = goals.find(goal => {
+      if (goal.goalId === goalId) {
+        return { completed: goal.completed };
+      }
```

Comment:
You should return something simple, like a boolean. You can just `return goal.goalId === goalId`, no `if` required.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find#description